### PR TITLE
sd-device-monitor: trivial cleanups

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #18717.

### CODE REVIEW

File: src/libsystemd/sd-device/device-monitor.c [hunk 1]
```diff
@@ -718,41 +718,30 @@ _public_ int sd_device_monitor_filter_update(sd_device_monitor *m) {
...
-
-        if (devtype) {
-                d = strdup(devtype);
-                if (!d)
-                        return -ENOMEM;
-        }
-
-        r = hashmap_ensure_put(&m->subsystem_filter, NULL, s, d);
-        if (r < 0)
+        r = hashmap_put_strdup_full(&m->subsystem_filter, &trivial_hash_ops_free_free, subsystem, devtype);
```
Comment (poettering) [inline src/libsystemd/sd-device/device-monitor.c:728]: shouldn't this be string_hash_ops_free_free?

Comment (yuwata) [reply src/libsystemd/sd-device/device-monitor.c:728]: No, it must be `trivial_hash_ops_free_free`. As user may set two or more different devtypes for the same subsystem. See #17259 and a0887abbd8bd9f1a9a975af08e6b4a43960bb3e2.

Comment (poettering) [reply src/libsystemd/sd-device/device-monitor.c:728]: could you add a comment about this here?

Comment (yuwata) [reply src/libsystemd/sd-device/device-monitor.c:728]: Sure, added.

**Pull request merged**